### ISSUE TITLE
[Feature request] Implement Nostr signer in the Keepassxc browser plugin

### ISSUE THREAD

**Issue by racuna** [keepassxreboot/keepassxc-browser#2388]
Hi,

Nostr is a protocol (Notes and other stuff transmitted by relays) that can be used for building social networks, blogs, reddit-like sites, and so on.

This protocol works using keys (public and private), and there is a NIP (Nostr Implementation Possibilities) that is used to provide a way to sign in to a Nostr platform without giving them the private key.

It would be interesting to see this implemented in Keepassxc instead of using another browser plugin to do it.

NIP-07 description: https://github.com/nostr-protocol/nips/blob/master/07.md

An implementation for Chrome: https://github.com/fiatjaf/nos2x

**Comment by varjolintu** [comment]
What prevents you to use a custom KPH attribute or Auto-Type for this?

**Comment by racuna** [comment]
That would give your private key to the different Nostr web clients. A nip-07 browser extension allows you to sign Nostr events on web-apps without having to give them your keys

https://github.com/aljazceru/awesome-nostr#nip-07-browser-extensions

**Comment by varjolintu** [comment]
I have no previous knowledge about the whole Nostr, but looking at the video here https://github.com/fiatjaf/nos2x I can see that you can enter the public or private key directly to the input field. The extension can already support scenarios like that. Isn't that a viable solution?

**Comment by varjolintu** [comment]
I have no previous knowledge about the whole Nostr, but looking at the video here https://github.com/fiatjaf/nos2x I can see that you can enter the public or private key directly to the input field. The extension can already support scenarios like that. Isn't that a viable solution?

**Comment by racuna** [comment]
Yes, you can. But isn't recommended to put the private key in the input field of services (web-apps) you don't know if the dev has good intentions or not. That's why there is the NIP-07 proposal of implementation.

**Comment by droidmonkey** [comment]
This feels like an alternate universe passkey

**Comment by varjolintu** [comment]
I'm gonna say no to this one.
